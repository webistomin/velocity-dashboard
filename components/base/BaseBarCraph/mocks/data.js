export const barGraphData = {
  labels: ['April', 'May', 'June'],
  datasets: [
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Comfort',
      lineTension: 0.4,
      backgroundColor: '#2E5BFF',
      data: [60, 65, 85],
    },
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Premium',
      backgroundColor: '#8C54FF',
      data: [70, 80, 90],
    },
  ],
};
