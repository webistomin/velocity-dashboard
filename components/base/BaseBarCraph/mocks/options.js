export const options = {
  responsive: true,
  maintainAspectRatio: false,
  scales: {
    yAxes: [
      {
        gridLines: {
          color: '#B0BAC9',
          borderDash: [3, 3],
        },
        ticks: {
          suggestedMin: 0,
          suggestedMax: 125,
          fontColor: '#B0BAC9',
        },
      },
    ],
    xAxes: [
      {
        gridLines: {
          color: '#B0BAC9',
          borderDash: [3, 3],
        },
        ticks: {
          fontColor: '#B0BAC9',
        },
      },
    ],
  },
  legend: {
    display: true,
    align: 'end',
    labels: {
      boxWidth: 10,
      usePointStyle: true,
      padding: 20,
      fontColor: '#B0BAC9',
      fontFamily: 'Rubik, system-ui, sans-serif',
    },
  },
  elements: {
    point: {
      radius: 5,
      hoverRadius: 4,
      borderWidth: 3,
      hoverBorderWidth: 3,
      backgroundColor: '#FFF',
    },
  },
};
