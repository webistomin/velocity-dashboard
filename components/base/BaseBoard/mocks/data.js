export const boardData = [
  {
    title: 'Backlog',
    color: 'blue',
    tasks: [
      {
        title: 'Spire1',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire2',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire3',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'In progress',
    color: 'purple',
    tasks: [
      {
        title: 'Spire5',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire6',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire7',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire8',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'Validate',
    color: 'darkturquoise',
    tasks: [
      {
        title: 'Spire9',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire10',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire11',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire12',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'Complete',
    color: 'yellow',
    tasks: [
      {
        title: 'Spire13',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire14',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire15',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire16',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
];
