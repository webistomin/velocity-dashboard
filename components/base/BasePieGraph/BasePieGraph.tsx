import { VueComponent } from 'types/vue-components';
import { Component, Prop } from 'nuxt-property-decorator';
import { ChartData, ChartOptions } from 'chart.js';
import { Pie } from 'vue-chartjs';

export interface IBasePieGraphProps {
  chartData: ChartData;
  options?: ChartOptions;
  fallbackText?: string;
}

@Component({
  name: 'BasePieGraph',
  extends: Pie,
})
export default class BasePieGraph extends VueComponent<IBasePieGraphProps> {
  @Prop()
  private readonly chartData!: IBasePieGraphProps['chartData'];

  @Prop()
  private readonly options!: IBasePieGraphProps['options'];

  @Prop({ default: 'Pie graph' })
  private readonly fallbackText: IBasePieGraphProps['fallbackText'];

  mounted(): void {
    // @ts-ignore
    this.renderChart(this.chartData, this.options);

    this.$nextTick(() => {
      const canvas = this.$refs.canvas as HTMLCanvasElement;
      canvas.setAttribute('aria-label', this.fallbackText || 'Pie graph');
      canvas.setAttribute('role', 'img');
    });
  }
}
