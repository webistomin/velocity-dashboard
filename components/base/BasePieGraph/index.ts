import BasePieGraph from './BasePieGraph';

export default BasePieGraph;
