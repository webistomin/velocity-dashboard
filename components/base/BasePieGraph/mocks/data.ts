export const pieData = {
  labels: ['Fully serviced', 'In service', 'Waiting', 'Service needed'],
  datasets: [
    {
      fill: true,
      weight: 5,
      hoverBorderWidth: 10,
      borderWidth: 0,
      hoverBorderColor: ['#F7C137', '#00C1D4', '#8C54FF', '#2E5BFF'],
      hoverBackgroundColor: ['#F7C137', '#00C1D4', '#8C54FF', '#2E5BFF'],
      backgroundColor: ['#F7C137', '#00C1D4', '#8C54FF', '#2E5BFF'],
      data: [57.8, 5, 20, 17.2],
    },
  ],
};
