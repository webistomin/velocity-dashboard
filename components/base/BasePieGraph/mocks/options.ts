export const pieOptions = {
  responsive: true,
  maintainAspectRatio: false,
  animation: {
    animateScale: true,
  },
  legend: {
    display: true,
    position: 'right',
    align: 'center',
    labels: {
      boxWidth: 10,
      usePointStyle: true,
      padding: 30,
      fontColor: '#2e384d',
    },
  },
};
