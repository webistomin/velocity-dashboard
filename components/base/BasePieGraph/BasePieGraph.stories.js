import { object, text } from '@storybook/addon-knobs';
import BasePieGraph from './BasePieGraph';
import { pieOptions } from './mocks/options';
import { pieData } from './mocks/data';

export default {
  title: 'BasePieGraph',
  parameters: {
    options: {
      showPanel: true,
      selectedPanel: 'storybook/knobs/panel',
      previewTabs: {
        actions: {
          hidden: true,
        },
      },
    },
  },
};

export const Default = () => ({
  components: { BasePieGraph },
  props: {
    chartData: {
      type: Object,
      default: object('chartData', pieData),
    },
    options: {
      type: Object,
      default: object('options', pieOptions),
    },
    fallbackText: {
      type: String,
      default: text('fallbackText', ''),
    },
  },
  template: `<base-pie-graph :chartData='this.chartData' :options='this.options' :fallbackText='this.fallbackText' />`,
});
