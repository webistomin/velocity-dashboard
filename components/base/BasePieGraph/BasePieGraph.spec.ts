import { mount } from '@vue/test-utils';

import { pieOptions } from './mocks/options';
import { pieData } from './mocks/data';
import BasePieGraph from './BasePieGraph';

describe('BasePieGraph', () => {
  const factory = (propsData = {}, componentOptions = {}) => {
    return mount(BasePieGraph, {
      propsData: {
        options: pieOptions,
        chartData: pieData,
        ...propsData,
      },
      ...componentOptions,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory();
    const html = wrapper.html();
    expect(html).toMatchSnapshot();
  });

  it('Set default a11y attributes', async () => {
    const wrapper = factory();
    const vm = wrapper.vm;
    await vm.$nextTick();

    const canvas = vm.$refs.canvas as HTMLCanvasElement;
    const role = canvas.getAttribute('role');
    const ariaLabel = canvas.getAttribute('aria-label');

    expect(role).toEqual('img');
    expect(ariaLabel).toEqual('Pie graph');
  });

  it('Set aria-label from props', async () => {
    const expectedAriaLabel = 'My brand new pie graph';
    const wrapper = factory({
      fallbackText: expectedAriaLabel,
    });
    const vm = wrapper.vm;
    await vm.$nextTick();

    const canvas = vm.$refs.canvas as HTMLCanvasElement;
    const ariaLabel = canvas.getAttribute('aria-label');

    expect(ariaLabel).toEqual(expectedAriaLabel);
  });
});
