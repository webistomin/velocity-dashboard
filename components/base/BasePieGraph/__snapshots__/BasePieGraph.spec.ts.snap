// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasePieGraph Matches snapshot 1`] = `"<div class=\\"\\"><canvas id=\\"pie-chart\\" width=\\"0\\" height=\\"0\\" style=\\"display: block;\\" class=\\"chartjs-render-monitor\\"></canvas></div>"`;
